### PR TITLE
[PyOV] Fix mypy after bump, general refactor

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
 - PEP 484 was updated to require the explicit use of `Optional` and `mypy` enforces this since version `0.980`.
 - General refactor of code to comply with other tools like flake8 or black. (run black on all files, thus big number of changed files)

### Tickets:
 - *...*

### CODE REVIEW

File: .github/workflows/py_checks.yml [hunk 1]
```diff
@@ -46,10 +46,10 @@ jobs:
         with:
           name: samples_diff
           path: samples_diff.diff
-      # IE Python API Flake code-style
-      #- name: Run flake8 on IE Python API
-      #  run: python -m flake8 ./ --config=setup.cfg
-      #  working-directory: src/bindings/python/src/compatibility/openvino
+      IE Python API Flake code-style
```
Comment (mitruska) [inline .github/workflows/py_checks.yml:49]: Shouldn't it be still a comment?

File: .github/workflows/py_checks.yml [hunk 2]
```diff
@@ -61,10 +61,10 @@ jobs:
         with:
           name: ie_python_diff
           path: ie_python_diff.diff
-      # nGraph Python API Flake code-style
-      #- name: Run flake8 on nGraph Python API
-      #  run: python -m flake8 ./src/compatibility/ngraph --config=setup.cfg
-      #  working-directory: src/bindings/python
+      nGraph Python API Flake code-style
```
Comment (mitruska) [inline .github/workflows/py_checks.yml:64]: Shouldn't it be still a comment?

File: .github/workflows/py_checks.yml [hunk 3]
```diff
@@ -76,10 +76,10 @@ jobs:
         with:
           name: pyngraph_diff
           path: pyngraph_diff.diff
-      # Python API 2.0 Flake code-style
-      #- name: Run flake8 on Python API 2.0
-      #  run: python -m flake8 ./src/openvino --config=setup.cfg
-      #  working-directory: src/bindings/python
+      Python API 2.0 Flake code-style
```
Comment (mitruska) [inline .github/workflows/py_checks.yml:79]: Shouldn't it be still a comment?

File: .github/workflows/py_checks.yml [hunk 4]
```diff
@@ -106,23 +106,23 @@ jobs:
...
-      #  working-directory: src/bindings/python/src/compatibility/openvino
-      # nGraph Python API mypy check
-      #- name: Run mypy on nGraph Python API
-      #  run: python -m mypy ./src/compatibility/ngraph --config-file ./setup.cfg
-      #  working-directory: src/bindings/python
-      # Python API 2.0 mypy check
-      #- name: Run mypy on Python API 2.0
-      #  run: python -m mypy ./src/openvino --config-file ./setup.cfg
-      #  working-directory: src/bindings/python
+      Python API 2.0 tests Flake code-style
```
Comment (mitruska) [inline .github/workflows/py_checks.yml:109]: Shouldn't it be still a comment?

File: .github/workflows/py_checks.yml [hunk 5]
```diff
@@ -106,23 +106,23 @@ jobs:
...
+      - name: Run flake8 on python tests
+        ignore lack of docs in tests
+       run: python -m flake8 tests/ --config=setup.cfg
+       working-directory: src/bindings/python
+      IE Python API mypy check
```
Comment (mitruska) [inline .github/workflows/py_checks.yml:114]: Shouldn't it be still a comment?

**Review by akuporos — APPROVED** [review]

**Review by p-wysocki — APPROVED** [review]

File: .github/workflows/py_checks.yml (hunk 1 above)
Comment (jiwaszki) [reply .github/workflows/py_checks.yml:49]: Yes, my bad.

**Review by mitruska — APPROVED** [review]

**Pull request merged**